Remove duplicate code in AutoTableLayout

Ahmad-S792 opened the pull request:
#### 4ad1a56cbc43e030f8bcfc82b2a7ef9360766025
<pre>
Remove duplicate code in AutoTableLayout
<a href="https://bugs.webkit.org/show_bug.cgi?id=291477">https://bugs.webkit.org/show_bug.cgi?id=291477</a>

Reviewed by Alan Baradlay.

Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/43c8dcdd065c790e1d34b8ed1fd3bb326f7aa16f">https://chromium.googlesource.com/chromium/src.git/+/43c8dcdd065c790e1d34b8ed1fd3bb326f7aa16f</a>

This patch introduces helper function `shrinkCellWidth` to deal with
all different length types rather than repeating code.

* Source/WebCore/rendering/AutoTableLayout.cpp:
(WebCore::AutoTableLayout::layout):
(WebCore::AutoTableLayout::shrinkCellWidth):
* Source/WebCore/rendering/AutoTableLayout.h:

Canonical link: <a href="https://commits.webkit.org/293748@main">https://commits.webkit.org/293748@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/48a0039bf69ed307b6966250bed8f73e19b63b7d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99821 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19466 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9743 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104946 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50399 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101862 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19771 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27902 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/75976 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/33072 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102828 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15066 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90132 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56343 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14872 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8119 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49771 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/84791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8204 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107307 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26932 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/19663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/84933 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27293 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86337 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84460 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21443 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29136 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6848 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/20737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26868 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32079 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26679 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29996 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28240 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/AutoTableLayout.cpp:
```diff
@@ -662,84 +662,14 @@ void AutoTableLayout::layout()
+    // Need to reduce cells with the following prioritization:
+    // This is basically the reverse of how we grew the cells.
+    if (available < 0)
+        shrinkCellWidth(LengthType::Auto, available);
```
this should read
`available = shrinkCellWidth(LengthType::Auto, available);`
(with  `float shrinkCellWidth(const LengthType&, float available);`)

Ahmad-S792 replied:
@alanbaradlay - Done. If you can have another look that it is what you had in your mind?

alanbaradlay commented on Source/WebCore/rendering/AutoTableLayout.cpp:
```diff
@@ -749,4 +679,32 @@ void AutoTableLayout::layout()
     m_table->setColumnPosition(m_table->columnPositions().size() - 1, pos);
 }
 
+float AutoTableLayout::shrinkCellWidth(const LengthType& lengthType, float& available)
```
`float AutoTableLayout::shrinkCellWidth(const LengthType& lengthType, float& available)
`
should read 
`float AutoTableLayout::shrinkCellWidth(const LengthType& lengthType, float available)
`
but look great otherwise! Thank you.

Ahmad-S792 replied:
Oh! forgot - & part. Will push update now. :-)

Ahmad-S792 replied:
Done.

Commit:
Remove duplicate code in AutoTableLayout
https://bugs.webkit.org/show_bug.cgi?id=291477

Reviewed by Alan Baradlay.

Merge: https://chromium.googlesource.com/chromium/src.git/+/43c8dcdd065c790e1d34b8ed1fd3bb326f7aa16f

This patch introduces helper function `shrinkCellWidth` to deal with
all different length types rather than repeating code.

* Source/WebCore/rendering/AutoTableLayout.cpp:
(WebCore::AutoTableLayout::layout):
(WebCore::AutoTableLayout::shrinkCellWidth):
* Source/WebCore/rendering/AutoTableLayout.h:

Canonical link: https://commits.webkit.org/293748@main
